### PR TITLE
AD-185413: .NET8.0 Upgrade

### PR DESCRIPTION
**Pull request by hnandra-DFE**

### CODE REVIEW

File: NCS.DSS.Diversity/APIDefinition/GenerateDiversitySwaggerDoc.cs [hunk 1]
```diff
@@ -14,27 +12,24 @@ public class GenerateDiversitySwaggerDoc
         public const string ApiDefinitionName = "API-Definition";
         public const string ApiDefRoute = ApiTitle + "/" + ApiDefinitionName;
         public const string ApiDescription = "New validations rules for new and updated records. Updated to allow retrieval of the full diversity record";
-        public const string ApiVersion = "2.0.0";
+        public const string ApiVersion = "4.0.0";
```
[the remark this answers is not in the document]
Comment (hnandra) [reply NCS.DSS.Diversity/APIDefinition/GenerateDiversitySwaggerDoc.cs:15]: The function app is version 4.0 now

File: NCS.DSS.Diversity/PostDiversityHttpTrigger/Function/PostDiversityHttpTrigger.cs [hunk 2]
```diff
@@ -60,45 +55,47 @@ public PostDiversityHttpTrigger(IResourceHelper resourceHelper, IPostDiversityHt
         [Response(HttpStatusCode = (int)HttpStatusCode.Conflict, Description = "Diversity Details already exists for customer", ShowSchema = false)]
         [Response(HttpStatusCode = 422, Description = "Diversity validation error(s)", ShowSchema = false)]
         [Display(Name = "Post", Description = "Ability to create a new diversity record for a given customer.")]
-        public async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "Customers/{customerId}/DiversityDetails")]
-            HttpRequest req, ILogger log, string customerId)
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "Customers/{customerId}/DiversityDetails")]
+            HttpRequest req, string customerId)
         {
+            req.EnableBuffering();
```
[the remark this answers is not in the document]
Comment (hnandra) [reply NCS.DSS.Diversity/PostDiversityHttpTrigger/Function/PostDiversityHttpTrigger.cs:61]: https://github.com/dotnet/aspnetcore/issues/21997

File: NCS.DSS.Diversity/Program.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    .ConfigureServices(services =>
+    {
+        services.AddApplicationInsightsTelemetryWorkerService();
+        services.ConfigureFunctionsApplicationInsights();
+        services.AddLogging();
+        services.AddSingleton<IResourceHelper, ResourceHelper>();
+        services.AddSingleton<IValidate, Validate>();
+        services.AddSingleton<ILoggerHelper, LoggerHelper>();
+        services.AddSingleton<IHttpRequestHelper, HttpRequestHelper>();
+        services.AddSingleton<IHttpResponseMessageHelper, HttpResponseMessageHelper>();
```
[the remark this answers is not in the document]
Comment (hnandra) [reply NCS.DSS.Diversity/Program.cs:30]: IHttpResponseMessageHelper removed.

File: Resources/deployments/function-app-resources.json [hunk 4]
```diff
@@ -68,6 +68,10 @@
             {
               "name": "FUNCTIONS_EXTENSION_VERSION",
               "value": "~4"
+            },
```
[the remark this answers is not in the document]
Comment (hnandra) [reply Resources/deployments/function-app-resources.json:71]: Updated netFrameworkVersion to v8.0

File: NCS.DSS.Diversity/Helpers/Helpers.cs [hunk 5]
```diff
@@ -1,23 +1,54 @@
 ﻿using Microsoft.AspNetCore.Http;
 using NCS.DSS.Diversity.Models;
 using Newtonsoft.Json;
```
[the remark this answers is not in the document]
Comment (hnandra) [reply NCS.DSS.Diversity/Helpers/Helpers.cs]: Fixed in commit https://github.com/SkillsFundingAgency/dss-diversity/pull/45/commits/9d6f9da19ccc36753706f1ae494cc1d2ac108c83 #

**Review by PradeepVoorukonda — APPROVED** [review]

**Review by gwilson-DFE — APPROVED** [review]

**Review by gwilson-DFE — APPROVED** [review]
Approved

**Review by gwilson-DFE — APPROVED** [review]

**Review by PradeepVoorukonda — DISMISSED** [review]

**Comment by bmatthews-DFE** [comment]
This PR has been replaced with: https://github.com/SkillsFundingAgency/dss-diversity/pull/46

**Pull request closed without merging**